addons/l10n_it_edi_sdicoop/tests/test_edi_xml.py

```diff
@@ -554,7 +554,8 @@ def test_non_latin_and_latin_invoice(self):
               </DettaglioLinee>
               <DatiRiepilogo>
                 <AliquotaIVA>22.00</AliquotaIVA>
-                <ImponibileImporto>2401.20</ImponibileImporto>
+                <Arrotondamento>0.02727273</Arrotondamento>
+                <ImponibileImporto>2401.23</ImponibileImporto>
```

Comment:
I find it weird to alter the base amount, while we are very certain that the base amount is 2401.30 in all cases...  @dbkosky

Reply:
In this case I think it's actually because the `Imposta` is wrong, probably due to local rounding
you would assume that it would be `800.4 * 3 * 0.22 = 528.264` which rounds to `528.26`
However here it is `528.27`,  because
`800.4 * 0.22 = 176.088` rounds to `176.09` which when multiplied by three is `528.27`

`528.27` is the tax we have in odoo.
Since the specs state that the ImportoTotale must be equal to `PrezzoTotale + ImportoContributoCassa + Arrotondamento`, and we still want to have an `Imposta` that reflects the value we have on the invoice, we have to adapt the `ImponibileImporto`.
This is because of the constraint
` (AliquotaIVA * ImponibileImporto / 100) = Imposta`. 

So we adapt the value of the `ImportoTotale`, and represent the difference using the `Arrotondamento`
`2401.23 * 0.22 = 528.2706000000001`

Reply:
I is weird, but "Arrotondamento" is there to acknowledge this base amount variation. According to specs, there is no way around it.

Reply:
And this way, taxes declared are exactly the taxes on the invoice witch is what matters ?

Reply:
@jco-odoo soooo r+ ? 😁

Reply:
@jbw-odoo I still have a problem with this because it changes the base amounts.  It would be good to already check if these cases would be solved by just putting a round globally.

Reply:
I think that in Italy one should compute taxes by summing the lines per each different tax amount.
As the e-Invoicing came along it came to be from an un-enforced law and custom, to be precisely checked for each invoice.

This is an example of invoice from an ERP. 
The table in the bottom right clearly shows how taxes are calculated.
![image](https://user-images.githubusercontent.com/1665365/197965863-55528720-43cc-4791-b9ef-6009ec37f064.png)

Here an user of the "developers that deal with Italian PA" forum also states that "some ERP calculate taxes by line, but that (the other user's calculation on the sum of the lines for the same tax amount) is more correct"
https://forum.italia.it/t/calcolo-importo-fattura-ricevuto/7592/3

I can't find an official reference, yet, but... we have clues.

Reply:
So, in the light of this, if you have an Italian company, then you must round globally.